pandas/core/indexing.py

```diff
@@ -1840,6 +1840,17 @@ def _setitem_single_column(self, loc: int, value, plane_indexer):
         """
         pi = plane_indexer
 
+        if not hasattr(self.obj._mgr, "blocks"):
```

Comment:
shouldn't this be an ArrayManager test? why the different semantics?

Reply:
This is just a check to know if it is an ArrayManager, without actually importing it (core/indexing.py currently doesn't import anything from internals). 
I think in another PR that might not have been merged, I added an attribute on the DataFrame that returns True/False for this, which might be easier to reuse in multiple places.

Reply:
yah we should for sure have 1 canonical way of doing this check so we can grep for all the places we do it

Reply:
-> https://github.com/pandas-dev/pandas/pull/44676